### PR TITLE
Added fail-fast null-checking in ParsedBuildbotData and ParsedGitData

### PR DESCRIPTION
**Pull request by MagoDelBlocco**
Removed the self-checking methods in ParsedBuildbotData and ParsedGitData, and instead moved the checks inside their constructors. This guarantees that no instance of them can have invalid fields, and thus requires less checks further down the application.

Now, their constructors throw IllegalArgumentException when a required field is null.

### CODE REVIEW

File: src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java [hunk 1]
```diff
@@ -43,25 +43,20 @@
    * @param logs logs for each individual stage of compilation. See {@link
    *      #com.google.graphgeckos.dashboard.storage.Log Log}
    * @param status the results of the compilation (true for passed, false for failed)
+   * @throws IllegalArgumentException if either parameter is null
    */
   public ParsedBuildbotData(String commitHash, String builderName,
-                            List<Log> logs, BuilderStatus status) {
+                            List<Log> logs, BuilderStatus status) throws IllegalArgumentException {
+    if (commitHash != null && builderName != null && logs != null && status != null) {
```
Comment (holounic) [inline src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:50]: I suppose, it should be ```(commitHash == null || builderName == null || logs == null || status == null)```

**Review by holounic — APPROVED** [review]
LGTM!

**Comment by holounic** [comment]
I would suggest to use [```@NonNull```](https://projectlombok.org/features/NonNull) to mark the args that shouldn't be null. This can make our code easier to read.

**Comment by MagoDelBlocco** [comment]
> I would suggest to use @NonNull to mark the args that shouldn't be null.

Will add that in a future PR. Thank you for the suggestion!

**Review by krasimir-google — APPROVED** [review]
This is great!

**Pull request merged**